grpc/src/main/java/com/replica/api/StreetRouter.java

```diff
@@ -36,6 +43,18 @@ public StreetRouter(GraphHopper graphHopper,
         this.customTags = customTags;
     }
 
+    public static Long calculatePathId(ResponsePath responsePath) {
+        return calculatePathId(getPathStableEdgeIds(responsePath), getEdgeTimes(responsePath));
+    }
+
+    public static Long calculatePathId(List<String> pathStableEdgeIds, List<Long> pathEdgeTimes) {
```

Comment:
Ah, thanks for the reminder here, and apologies that I totally forgot that we'd had this conversation already! I'm completely on board with using `PointList`, it's definitely cleaner (esp with the built-in functions it provides that you mention). And good point about edge time not being appropriate here too, I agree.

I've updated things to rely on `PointList.hashcode` as our "path ID". One minor detail to note - I implemented the deduping by relying on the hashcodes only (ie storing a set of hashes + comparing against that set when deduping), as opposed to storing a set of `PointList`s and using that class's `equals` to determine route equality. 

Does this seem reasonable to you? It looks like `equals` might be a slightly more rigorous check than simply comparing hashes, but the probability of non-identical routes having matching hashes in this context seems fairly low. Happy to switch to a more formal `PointList` equality-based deduping scheme if we're worried about that case, though!

Reply:
Hmm yeah I think storing the hashcodes themselves in a set is kinda roundabout. Java HashSets already use the hashcode of the entries to determine if a given element is already present. The hashcodes help optimize performance, but the sets tolerate hash collisions and use `equals` for further deduping when necessary. This design is pretty general and is used across languages - https://en.wikipedia.org/wiki/Hash_table

So I think it would be more natural (and robust, in the event of hash collisions!) to replace `Set<Integer> pathIdsInReturnSet` with something like `Set<PointList> pathsInReturnSet`. I think we can remove the concept of a "path id" and just require that `!pathsInReturnSet.contains(responsePath.getPoints())` before including the response path in the StreetRouteReply.

Reply:
Makes perfect sense! Thanks for the reminder re: hashsets' implementation under-the-hood (and apologies for what I'm realizing was a somewhat lazy question...should've thought through that a bit more before hitting send 🤦).

I've torn out all mentions of "path ID" and made the logic updates in the server to use the `PointList` in place of the hashcode/path ID, and I removed `testCalculatePathId` altogether, as it's not really relevant anymore